utils/check_ptc.py

```diff
@@ -0,0 +1,89 @@
+""" Tool for sanity checking rrgraph CHAN PTC's.
+"""
+import lxml.etree as ET
+import argparse
+
+
+def check_ptc(xml):
+    """ Checks ptc values used on CHANX and CHANY rr graph nodes are valid.
+
+    CHAN ptc numbers are an index per x/y coordinate and channel type (CHANX or
+    CHANY).  ptc's at a particular coordinate/type must start at 0, and fill
+    to the max value.
+
+    """
+    chan_ptcs = {}
+    nodes = {}
+
+    for node in xml.find('rr_nodes').iter('node'):
+        assert node.attrib['id'] not in nodes
+        nodes[node.attrib['id']] = node
+
+        node_type = node.attrib['type']
+
+        if node_type in ['CHANX', 'CHANY']:
+            loc_xml = node.find('loc')
+            assert loc_xml is not None
+
+            for x in range(int(loc_xml.attrib['xlow']),
+                           int(loc_xml.attrib['xhigh']) + 1):
+                for y in range(int(loc_xml.attrib['ylow']),
+                               int(loc_xml.attrib['yhigh']) + 1):
+                    key = (node_type, x, y)
+
+                    if key not in chan_ptcs:
+                        chan_ptcs[key] = []
+
+                    chan_ptcs[key].append(
+                        (node.attrib['id'], int(loc_xml.attrib['ptc']))
+                    )
+
+    for (node_type, x, y), node_ptcs in chan_ptcs.items():
+        nodes, ptcs = zip(*node_ptcs)
+        starts_at_zero = min(ptcs) == 0
+        ends_at_max_val = max(ptcs) == len(ptcs) - 1
+        all_values_present = len(ptcs) == len(set(ptcs))
+
+        if not all((starts_at_zero, ends_at_max_val, all_values_present)):
+            sorted_nodes = sorted(zip(ptcs, nodes), key=lambda x: x[0])
+            for idx, (ptc, node) in enumerate(sorted_nodes):
+                if idx != ptc:
```

Comment:
Can you save a level of indenting with;
```suggestion
                if idx == ptc:
                    continue
```